modules/tinymce/src/plugins/table/main/ts/api/Options.ts

```diff
@@ -18,21 +18,41 @@ const defaultCellBorderStyles = Arr.map([ 'Solid', 'Dotted', 'Dashed', 'Double',
   return { title: type, value: type.toLowerCase() };
 });
 
+const stylesAvailableAsAttributes = [ 'width' ];
+
+const getPixelForcedWidth = (editor: Editor) => {
+  // Determine the inner size of the parent block element where the table will be inserted
+  const dom = editor.dom;
+  const parentBlock = dom.getParent<HTMLElement>(editor.selection.getStart(), dom.isBlock) ?? editor.getBody();
+  return Width.getInner(SugarElement.fromDom(parentBlock)) + 'px';
+};
+
 // Note: This is also contained in the core Options.ts file
 const determineDefaultStyles = (editor: Editor, defaultStyles: Record<string, string>): Record<string, string> => {
-  if (isPixelsForced(editor)) {
-    // Determine the inner size of the parent block element where the table will be inserted
-    const dom = editor.dom;
-    const parentBlock = dom.getParent<HTMLElement>(editor.selection.getStart(), dom.isBlock) ?? editor.getBody();
-    const contentWidth = Width.getInner(SugarElement.fromDom(parentBlock));
-    return { ...defaultStyles, width: contentWidth + 'px' };
-  } else if (isResponsiveForced(editor)) {
-    return Obj.filter(defaultStyles, (_value, key) => key !== 'width');
+  if (isResponsiveForced(editor) || !shouldStyleWithCss(editor)) {
```

Comment:
As hinted in the `Note` above, if we are making changes to this function, they will also have to be made in `modules/tinymce/src/models/dom/main/ts/table/api/Options.ts`